enigma/src/main/java/cuchaz/enigma/translation/mapping/MappingValidator.java

```diff
@@ -35,45 +39,105 @@ public void validateRename(ValidationContext vc, Entry<?> entry, String name) {
 
 	private void validateUnique(ValidationContext vc, Entry<?> entry, String name) {
 		ClassEntry containingClass = entry.getContainingClass();
-		Collection<ClassEntry> relatedClasses = getRelatedClasses(containingClass);
-
-		for (ClassEntry relatedClass : relatedClasses) {
-			Entry<?> relatedEntry = entry.replaceAncestor(containingClass, relatedClass);
-			Entry<?> translatedEntry = deobfuscator.translate(relatedEntry);
-
-			Collection<Entry<?>> translatedSiblings = obfToDeobf.getSiblings(relatedEntry).stream()
-					.map(deobfuscator::translate)
-					.collect(Collectors.toList());
+		InheritanceIndex inheritanceIndex = index.getInheritanceIndex();
 
-			if (!isUnique(translatedEntry, translatedSiblings, name)) {
-				Entry<?> parent = translatedEntry.getParent();
+		//Step 1, check it's unique within its own siblings
+		Collection<Entry<?>> directTranslatedSiblings = obfToDeobf.getSiblings(entry).stream()
+				.map(deobfuscator::translate)
+				.collect(Collectors.toList());
+		for (Entry<?> sibling : directTranslatedSiblings) {
+			if (entry.canConflictWith(sibling) && sibling.getName().equals(name) && isSynthetic(entry) == isSynthetic(sibling)) {
```

Comment:
Not entirely sure on the synthetic check - I wasn't able to reproduce it with the unit test, only with an SRG named MC jar (and an unrelated change I removed from this PR as it was probably broken).

The idea is that if one entry is a synthetic/bridge and the other is not, that we can safely accept the rename (as it's probably only the return type that differs)

Reply:
I think that should be `[..] && !isSynthetic(entry) && !isSynthetic(sibling)`, since if any of the methods is synthetic, they don't conflict since at least one of them is invisible from source code